File: codegen-test/src/test/java/com/alibaba/fastjson2/internal/processor/collections/JSONArrayTest.java
package com.alibaba.fastjson2.internal.processor.collections;

import com.alibaba.fastjson2.JSON;
import com.alibaba.fastjson2.JSONArray;
import com.alibaba.fastjson2.annotation.JSONCompiled;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class JSONArrayTest {
    @Test
    public void test() {
        Bean bean = new Bean();
        bean.v01 = new JSONArray();

        String str = JSON.toJSONString(bean);
        Bean bean1 = JSON.parseObject(str, Bean.class);
        assertEquals(bean.v01, bean1.v01);
    }

    @JSONCompiled
    public static class Bean {
        public JSONArray v01;
    }

    @Test
    public void test1() {
        Bean1 bean = new Bean1();
        bean.v01 = new JSONArray();

        String str = JSON.toJSONString(bean);
        Bean1 bean1 = JSON.parseObject(str, Bean1.class);
        assertEquals(bean.v01, bean1.v01);
    }

    @JSONCompiled(referenceDetect = false)
    public static class Bean1 {
        public JSONArray v01;
    }
}

File: kotlin/src/test/kotlin/com/alibaba/fastjson2/JSONArrayTest.kt
package com.alibaba.fastjson2

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class JSONArrayTest {
    @Test
    fun test_getObject() {
        // JSONArray
        val data = """[{"id":1,"name":"kraity"}]""".parseArray()

        val user = data.to<User>(0)
        assertEquals(1, user.id)
        assertEquals("kraity", user.name)
    }

    @Test
    fun test_toObject() {
        // JSONArray
        val data = "[{\"id\": 1, \"name\": \"fastjson\"}, {\"id\": 2, \"name\": \"fastjson2\"}]".parseArray()

        val list = data.toList<User>()
        assertEquals(2, list.size)

        list[0].apply {
            assertEquals(1, id)
            assertEquals("fastjson", name)
        }

        list[1].apply {
            assertEquals(2, id)
            assertEquals("fastjson2", name)
        }
    }

    data class User(
        val id: Int,
        val name: String
    )
}
